commands/modmail.py

```diff
@@ -0,0 +1,161 @@
+from discord.commands import Option
+from discord.enums import SlashCommandOptionType
+from formatting.embed import gen_embed, embed_splitter
+from __main__ import log, db
+
+
+# Define a simple View that gives us a confirmation menu
+class Confirm(discord.ui.View):
+    def __init__(self):
+        super().__init__()
+        self.value = None
+
+    # When the confirm button is pressed, set the inner value to `True` and
+    # stop the View from listening to more input.
+    # We also send the user an ephemeral message that we're confirming their choice.
+    @discord.ui.button(label="Yes", style=discord.ButtonStyle.green)
+    async def confirm(self, button: discord.ui.Button, interaction: discord.Interaction):
+        await interaction.response.send_message("Confirming", ephemeral=True)
+        for item in self.children:
+            item.disabled = True
+        self.value = True
+        self.stop()
+
+    # This one is similar to the confirmation button except sets the inner value to `False`
+    @discord.ui.button(label="No", style=discord.ButtonStyle.red)
+    async def cancel(self, button: discord.ui.Button, interaction: discord.Interaction):
+        log.info('Workflow cancelled')
+        await interaction.response.send_message("Modmail cancelled.", ephemeral=True)
+        for item in self.children:
+            item.disabled = True
+        self.value = False
+        self.stop()
+
+
+class Modmail(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def modmail_prompt(self, interaction: discord.Interaction):
+        listen_channel = interaction.channel
+
+        def check(m):
+            return m.author == interaction.user and m.channel == listen_channel
+
+        await interaction.response.send_message(embed=gen_embed(title='Modmail Message Contents',
+                                                  content='Please type out your modmail below and send. You can send images by adding an attachement to the message you send.'))
+
+        try:
+            mmsg = await self.bot.wait_for('message', check=check, timeout=300.0)
+        except asyncio.TimeoutError:
+            await interaction.channel.send(embed=gen_embed(title='Modmail Cancelled',
+                                                      content='The modmail has been cancelled.'))
+            return None
+        return mmsg
+
+    @discord.slash_command(name='modmail',
+                           description='Start a modmail. A modmail channel must be configured first.\nUse %serverconfig modmail [channel]')
+    async def modmail(self,
+                      ctx: discord.ApplicationContext,
+                      recipient: Option(discord.User, "User to send modmail to")):
+        modmail_content = await self.modmail_prompt(ctx.interaction)
+
+        if modmail_content:
+            log.info('Message detected from user')
+            view = Confirm()
+            sent_message = await ctx.interaction.channel.send(
```

Comment:
Best practice is to use the webhook for followup messages

```suggestion
            sent_message = await ctx.interaction.followup.send(
```